src/test/java/org/ergoplatform/ErgoAddressEncoderSpec.scala

```diff
@@ -0,0 +1,28 @@
+package org.ergoplatform
+
+import java.math.BigInteger
+
+import org.scalatest.{Matchers, PropSpec, TryValues}
+import scapi.sigma.DLogProtocol
+import scapi.sigma.DLogProtocol.DLogProverInput
+import sigmastate.Values
+import sigmastate.serialization.ValueSerializer
+
+class ErgoAddressEncoderSpec extends PropSpec with Matchers with TryValues {
+
+  property("scriptToAddress() should properly distinct all types of addresses from plain script bytes") {
```

Comment:
@oskin1 I've sneaked in `ErgoAddressSpecification` from ergo yesterday after your PR. It's in the master already. Could you please move this test there. And I suppose it's `fromProposition()` instead of `scriptToAddress()`.

Reply:
@greenhat, done